src/main/java/seedu/module/model/module/Deadline.java

```diff
@@ -2,25 +2,81 @@
 
 import static java.util.Objects.requireNonNull;
 
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.EnumSet;
+
+import seedu.module.logic.commands.exceptions.CommandException;
+import seedu.module.logic.parser.exceptions.ParseException;
+
+/**
+ * Represents Priority tags in Enum
+ */
+enum Priority {
+    HIGH, MEDIUM, LOW;
+    public boolean isValid() {
+        return true;
+    }
+}
+
 /**
  * Represents a Module's deadline in the module book.
  * Guarantees: immutable; is always valid
  */
 public class Deadline {
     public static final String MESSAGE_CONSTRAINTS = "Not a valid Deadline";
+    protected Date date;
 
     private String description;
     private String time;
     private boolean isDone;
     private boolean isInProgress;
     private String tag;
-    private int deadlineListNum = 0;
 
-    public Deadline(String description, String time, String tag) {
+    public Deadline (String description, String time, String tag) throws ParseException, CommandException {
         requireNonNull(description);
+        int maxLength = 70;
+        if (description.length() > maxLength) {
+            description = description.substring(0, maxLength);
+        }
         this.description = description;
         this.time = time;
-        this.tag = tag;
+        if (!isValidPriority(tag)) {
+            this.tag = tag;
+        } else {
+            throw new CommandException("Not a valid priority tag");
+        }
+        if (isDateValid(time)) {
+            this.date = parseDate(time);
+        } else {
+            throw new CommandException("Date and time not in dd/MM/yyyy HHmm format");
+        }
+    }
+
+    /**
+     * Validates priority tag.
+     * @param tag priority tag.
+     * @return true if input tag is valid, false otherwise.
+     */
+    public boolean isValidPriority(String tag) {
+        EnumSet<Priority> except = EnumSet.of(Priority.HIGH, Priority.MEDIUM, Priority.LOW);
+        return !except.contains(Priority.valueOf(tag));
+    }
```

Comment:
[MAJOR] Maybe an easier implementation is to convert the string to the corresponding enum using `Priority.valueOf(String)` method, which throws `IllegalArgumentException` if the string does not convert to any of the enums. This way we can achieve type safety by storing `tag` as a `Priority` enum.

Reference: https://stackoverflow.com/questions/604424/how-to-get-an-enum-value-from-a-string-value-in-java